lib/streamlit/type_util.py

```diff
@@ -335,6 +342,22 @@ def is_pandas_version_less_than(v: str) -> bool:
     return version.parse(pd.__version__) < version.parse(v)
 
 
+def pyarrow_table_to_bytes(table: pa.Table) -> bytes:
+    """Convert pyarrow.Table to bytes.
```

Comment:
Nit - let's specify what _sort_ of bytes these are: "Serialize pyarrow.Table to bytes using Apache Arrow"

Reply:
In the current version of Streamlit(v0.84), we don't support pyarrow tables. 

For the sake of continuity, we should throw an error if a user tries to use pyarrow table with `config.dataFrameSerialization = "legacy"` with the following error message. (Same message can go in the docstrings as well if needed)

> pyarrow tables are not supported with the legacy implementation, i.e. with `config.dataFrameSerialization = "legacy"`. 
> 
> To be able to use pyarrow tables, please enable pyarrow by changing the config setting, `config.dataFrameSerialization = "arrow"` 



cc: @tvst